Use docker client TLS config for getting bearer token

linkvt opened the pull request:
Fixes #654 

I also removed the `TODO` as we use the parameters value now which is set on the `dockerClient`s TLS config.
I don't know what runcom means or if this was forced to true, so I would be thankful for some explanations if I missed something there.
The fix is quite trivial as it is already done like this a few lines below:
https://github.com/containers/image/blob/master/docker/docker_client.go#L560
By looking at the blame it seems like this was just forgotten as the old line was 3 years old.

I tested logging in with podman, logging out and reusing an existing login from `docker`:
```
 vincent@vincent-pc  libpod   master ●  bin/podman logout docker-reg --log-level debug
INFO[0000] running as rootless                          
DEBU[0000] Looking for TLS certificates and private keys in /etc/containers/certs.d/docker-reg 
DEBU[0000]  crt: /etc/containers/certs.d/docker-reg/ca.crt 
DEBU[0000]  cert: /etc/containers/certs.d/docker-reg/client.cert 
DEBU[0000]  key: /etc/containers/certs.d/docker-reg/client.key 
DEBU[0000] GET https://docker-reg/v2/           
DEBU[0000] Ping https://docker-reg/v2/ status 401 
DEBU[0000] GET https://docker-reg/artifactory/api/docker/null/v2/token?service=docker-reg 
Not logged into docker-reg
 vincent@vincent-pc  libpod   master ●  bin/podman login docker-reg --log-level debug
INFO[0000] running as rootless                          
Username: uid
Password: 
DEBU[0004] Looking for TLS certificates and private keys in /etc/containers/certs.d/docker-reg 
DEBU[0004]  crt: /etc/containers/certs.d/docker-reg/ca.crt 
DEBU[0004]  cert: /etc/containers/certs.d/docker-reg/client.cert 
DEBU[0004]  key: /etc/containers/certs.d/docker-reg/client.key 
DEBU[0004] GET https://docker-reg/v2/           
DEBU[0004] Ping https://docker-reg/v2/ status 401 
DEBU[0004] GET https://docker-reg/artifactory/api/docker/null/v2/token?account=uid&service=docker-reg 
DEBU[0004] GET https://docker-reg/v2/           
Login Succeeded!
 vincent@vincent-pc  libpod   master ●  bin/podman logout docker-reg --log-level debug
INFO[0000] running as rootless                          
Removed login credentials for docker-reg
 vincent@vincent-pc  libpod   master ●  docker login docker-reg                  
Username: uid
Password: 
Error response from daemon: Get https://docker-reg/v2/: remote error: tls: handshake failure
 ✘ vincent@vincent-pc  libpod   master ●  docker login docker-reg
Username: uid
Password: 
WARNING! Your password will be stored unencrypted in /home/vincent/.docker/config.json.
Configure a credential helper to remove this warning. See
https://docs.docker.com/engine/reference/commandline/login/#credentials-store

Login Succeeded
 vincent@vincent-pc  libpod   master ●  bin/podman login docker-reg --log-level debug 
INFO[0000] running as rootless                          
Authenticating with existing credentials...
DEBU[0000] Looking for TLS certificates and private keys in /etc/containers/certs.d/docker-reg 
DEBU[0000]  crt: /etc/containers/certs.d/docker-reg/ca.crt 
DEBU[0000]  cert: /etc/containers/certs.d/docker-reg/client.cert 
DEBU[0000]  key: /etc/containers/certs.d/docker-reg/client.key 
DEBU[0000] GET https://docker-reg/v2/           
DEBU[0000] Ping https://docker-reg/v2/ status 401 
DEBU[0000] GET https://docker-reg/artifactory/api/docker/null/v2/token?account=uid&service=docker-reg 
DEBU[0000] GET https://docker-reg/v2/           
Existing credentials are valid. Already logged in to docker-reg
```

TomSweeneyRedHat commented:
LGTM, but definitely want @mtrmac to look and maybe @runcom if we can steal his eyes for a bit.

linkvt commented:
Thank you for your feedback, I will verify that.

linkvt commented:
@mtrmac yes it works, as you can see below I tried with the current source, switched to your branch and switched back:
```
 vincent@vincent-pc  libpod   master  go list -m all | grep image                                                                                    
github.com/containers/image v2.0.1+incompatible
github.com/opencontainers/image-spec v1.0.1
github.com/openshift/imagebuilder v1.1.0
golang.org/x/image v0.0.0-20190622003408-7e034cad6442
 vincent@vincent-pc  libpod   master  make podman                
...
 vincent@vincent-pc  libpod   master  bin/podman login docker-reg --log-level debug
...
ERRO[0004] error authenticating creds for "docker-reg": Get https://docker-reg/artifactory/api/docker/null/v2/token?account=uid&service=docker-reg: remote error: tls: handshake failure 
 ✘ vincent@vincent-pc  libpod   master  go mod edit -replace github.com/containers/image=github.com/mtrmac/image@bearer-token-tls                      
 vincent@vincent-pc  libpod   master ●  go mod vendor                                                                            
go: finding github.com/mtrmac/image bearer-token-tls
 vincent@vincent-pc  libpod   master ●  make podman                                                                              
...
 vincent@vincent-pc  libpod   master ●  bin/podman login docker-reg --log-level debug                                    
...
Login Succeeded!
 vincent@vincent-pc  libpod   master ●  git rev .   # resets all changes
 vincent@vincent-pc  libpod   master  go mod vendor
 vincent@vincent-pc  libpod   master  make podman                                          
...
 vincent@vincent-pc  libpod   master  bin/podman login docker-reg --log-level debug
...
ERRO[0004] error authenticating creds for "docker-reg": Get https://docker-reg/artifactory/api/docker/null/v2/token?account=uid&service=docker-reg: remote error: tls: handshake failure 
```

As your solution is better of course feel free to close my PR and merge yours.

rhatdan commented:
@mtrmac Can we merge this before we cut 3.0.0?

linkvt commented:
Closing this PR as the alternative was merged in https://github.com/containers/image/pull/669.
Thanks!
